airgun/entities/errata.py

```diff
@@ -0,0 +1,157 @@
+        :rtype: list
+        """
+        view = self.navigate_to(self, 'All')
+        return view.search(
+            value,
+            applicable=applicable,
+            installable=installable,
+            repo=repo
+        )
+
+    def read(self, entity_name, applicable=False, installable=False, repo=None,
+             environment=None):
+        """Read errata details.
+
+        :param str entity_name: errata id or title
+        :param bool applicable: filter by only applicable errata
+        :param bool installable: filter by only installable errata
+        :param str optional repo: filter by repository name
+        :param str optional environment: filter applicable hosts by environment
+            name
+        :return: dict representing tabs, with nested dicts representing fields
+            and values
+        :rtype: dict
+        """
+        view = self.navigate_to(
+            self,
+            'Details',
+            entity_name=entity_name,
+            applicable=applicable,
+            installable=installable,
+            repo=repo,
+        )
+        if environment:
+            view.content_hosts.environment_filter.fill(environment)
+        return view.read()
+
+    def install(self, entity_name, host_name):
+        """Install errata on content host.
+
+        :param str entity_name: errata id or title
+        :param str host_name: content host name to apply errata on
+        """
+        view = self.navigate_to(
+            self,
+            'Install',
+            entity_name=entity_name,
+            host_name=host_name,
+            applicable=False,
+            installable=False,
+            repo=None,
+        )
+        view.progressbar.wait_for_result()
+        return view.read()
+
+
+@navigator.register(ErrataEntity, 'All')
+class ShowAllErratum(NavigateStep):
+    """Navigate to All Erratum screen."""
+    VIEW = ErratumView
+
+    def step(self, *args, **kwargs):
+        self.view.menu.select('Content', 'Errata')
+
+
+@navigator.register(ErrataEntity, 'Details')
+class ErrataDetails(NavigateStep):
+    """Navigate to Errata details page.
+
+        Args:
+            entity_name: id or title of errata
+
+        Optional Args:
+            applicable: whether to filter errata by only applicable ones
+            installable: whether to filter errata by only installable ones
+            repo: name of repository to filter errata by
+    """
+    VIEW = ErrataDetailsView
+
+    def prerequisite(self, *args, **kwargs):
+        return self.navigate_to(self.obj, 'All')
+
+    def step(self, *args, **kwargs):
+        entity_name = kwargs.get('entity_name')
+        applicable = kwargs.get('applicable')
+        installable = kwargs.get('installable')
+        repo = kwargs.get('repo')
+        self.parent.search(
+            entity_name,
+            applicable=applicable,
+            installable=installable,
+            repo=repo
+        )
+        row_filter = {'title': entity_name}
```

Comment:
I feel some kind of redundancy here as your search already knew the criteria it searched with

Reply:
Yeah, but here i select different column depending on that criteria... Any better ideas? Theoretically we could try picking just first row available with no filtering, since searching was done and there shouldn't be lots of results, but i'm not sure that's a good idea.